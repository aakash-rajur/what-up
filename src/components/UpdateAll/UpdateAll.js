import PropTypes from "prop-types";
import React from "react";
import CHEVRON_DOWN_ICON from "../../assets/chevron-down.svg";
import {withUpdateAll} from "../../graphql/hoc";

export function UpdateAll({className, updateAll}) {
  return (
    <img
      src={CHEVRON_DOWN_ICON}
      alt="toggle-all"
      className={className}
      onClick={updateAll}
      title="Update All"
    />
  );
}

UpdateAll.propTypes = {
  className: PropTypes.string,
  nextStatus: PropTypes.string,
  filter: PropTypes.string
};

export default withUpdateAll(UpdateAll);
