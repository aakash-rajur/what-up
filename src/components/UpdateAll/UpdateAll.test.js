import React from 'react';
import CHEVRON_DOWN_ICON from "../../assets/chevron-down.svg";
import {UpdateAll} from './UpdateAll';

describe('UpdateAll component', () => {
	let className = 'lorem',
		onUpdateAll = stub(),
		component = shallow(<UpdateAll className={className} updateAll={onUpdateAll}/>);
	
	it('should render', () => {
		expect(component.find('img.lorem').exists())
			.toBe(true);
	});
	
	it('image source', () => Object.entries({
			src: CHEVRON_DOWN_ICON,
			alt: 'complete-all',
			className,
			onClick: onUpdateAll,
			title: 'Update All'
		}).forEach(([key, value]) =>
			expect(component.prop(key))
				.toBe(value)
		)
	);
	
	it('should invoke onUpdateAll', () => {
		component.simulate('click');
		expect(onUpdateAll.called).toBe(true);
		expect(onUpdateAll.callCount).toBe(1);
	});
	
	it('match snapshot', () => expect(component).toMatchSnapshot());
});
