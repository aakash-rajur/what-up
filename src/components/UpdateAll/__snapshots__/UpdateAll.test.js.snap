// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateAll component match snapshot 1`] = `
<img
  alt="complete-all"
  className="lorem"
  onClick={[Function]}
  src="chevron-down.svg"
  title="Update All"
/>
`;
